bot/code_review_bot/revisions.py

```diff
@@ -134,12 +134,33 @@ def __init__(
 
     @property
     def namespaces(self):
-        return [
-            f"phabricator.{self.phabricator_id}",
-            f"phabricator.diff.{self.diff_id}",
-            f"phabricator.phabricator_phid.{self.phabricator_phid}",
-            f"phabricator.diffphid.{self.diff_phid}",
-        ]
+        # Simplify repository names
+        def repo_slug(url):
+            if url.startswith("https://hg.mozilla.org/"):
+                url = url[23:]
+            return url.replace("/", "-")
+
+        out = []
+
+        # Phabricator indexes
+        if self.phabricator_id:
+            out.append(f"phabricator.{self.phabricator_id}")
+        if self.diff_id:
+            out.append(f"phabricator.diff.{self.diff_id}")
+        if self.phabricator_phid:
+            out.append(f"phabricator.phabricator_phid.{self.phabricator_phid}")
+        if self.diff_phid:
+            out.append(f"phabricator.diffphid.{self.diff_phid}")
+
+        # Revision indexes
+        if self.head_repository and self.head_changeset:
+            repo = repo_slug(self.head_repository)
+            out.append(f"head_repo.{repo}.{self.head_changeset}")
+        if self.base_repository and self.base_changeset:
+            repo = repo_slug(self.base_repository)
+            out.append(f"base_repo.{repo}.{self.base_changeset}")
```

Comment:
Will this be useful? Multiple different Phabricator revisions will update this index with their task ID.

Reply:
This modifies `namespaces`, a list of strings which gets used [here](https://github.com/mozilla/code-review/blob/b48a3e6bd380a854c6c6477873b89186241c8fee/bot/code_review_bot/workflow.py#L462).

```
            self.index_service.insertTask(
                namespace,
                {
                    "taskId": settings.taskcluster.task_id,
                    "rank": 0,
                    "data": payload,
                    "expires": stringDate(now + timedelta(days=TASKCLUSTER_INDEX_TTL)),
                },
            )
```

If `base_repo.{repo}.{self.base_changeset}` links to the last task which used it as a base revision, is there a good use for it?